x-pack/platform/plugins/shared/inference/common/output/bind_output.ts

```diff
@@ -5,31 +5,15 @@
  * 2.0.
  */
 
-import type {
-  OutputAPI,
-  OutputOptions,
-  BoundOutputAPI,
-  BoundOutputOptions,
-  UnboundOutputOptions,
-  ToolSchema,
-} from '@kbn/inference-common';
+import type { BoundOutputAPI, BoundOptions, OutputAPI } from '@kbn/inference-common';
+import { bindApi } from '@kbn/inference-common';
 
 /**
  * Bind output to the provided parameters,
  * returning a bound version of the API.
  */
-export function bindOutput(
-  chatComplete: OutputAPI,
-  boundParams: BoundOutputOptions
-): BoundOutputAPI;
-export function bindOutput(chatComplete: OutputAPI, boundParams: BoundOutputOptions) {
-  const { connectorId, functionCalling } = boundParams;
-  return (unboundParams: UnboundOutputOptions<string, ToolSchema, boolean>) => {
-    const params: OutputOptions<string, ToolSchema, boolean> = {
-      ...unboundParams,
-      connectorId,
-      functionCalling,
-    };
-    return chatComplete(params);
-  };
+export function bindOutput(outputApi: OutputAPI, boundParams: BoundOptions): BoundOutputAPI;
+
+export function bindOutput(outputApi: OutputAPI, boundParams: BoundOptions) {
```

Comment:
do we need overloading here? 
```suggestion
export function bindOutput(outputApi: OutputAPI, boundParams: BoundOptions): BoundOutputAPI {
```